ktor-core/src/org/jetbrains/ktor/features/CallLogging.kt

```diff
@@ -6,19 +6,22 @@ import org.jetbrains.ktor.pipeline.*
 import org.jetbrains.ktor.request.*
 import org.jetbrains.ktor.util.*
 import org.slf4j.*
+import org.slf4j.event.*
 
-class CallLogging(private val log: Logger, private val monitor: ApplicationMonitor) {
+class CallLogging(private val log: Logger, private val monitor: ApplicationMonitor, private val configuration: Configuration) {
```

Comment:
Do not store configuration in a feature instance. It is a mutable class that could potentially be captured from a config script and modified concurrently. Get a `level` property from it and store it instead.